protocol/communities/persistence.go

```diff
@@ -1698,3 +1700,53 @@ func (p *Persistence) RemoveAllCommunityRequestsToJoinWithRevealedAddressesExcep
 		WHERE community_id = ? AND public_key != ?;`, communityID, pk, communityID, pk)
 	return err
 }
+
+func (p *Persistence) SaveCommunityShard(communityID types.HexBytes, shard *shard.Shard, clock uint64) error {
+	var dbClock uint64
+	// Fetch existing Community Shard
+	err := p.db.QueryRow(`SELECT clock FROM communities_shards WHERE community_id = ? `, communityID).Scan(&dbClock)
```

Comment:
suggestion: add select and insert queries in one transaction

Reply:
done